src/librustc_typeck/check/method/suggest.rs

```diff
@@ -69,13 +69,27 @@ pub fn report_error<'a, 'tcx>(fcx: &FnCtxt<'a, 'tcx>,
             }
 
             if !static_sources.is_empty() {
-                fcx.tcx().sess.fileline_note(
+                cx.sess.fileline_note(
                     span,
                     "found defined static methods, maybe a `self` is missing?");
 
                 report_candidates(fcx, span, item_name, static_sources);
             }
 
+            if !predicates.is_empty() {
+                let bound_list = predicates.iter()
+                    .map(|p| format!("`{} : {}`",
+                                     p.substs.self_ty().unwrap(),
+                                     p))
+                    .collect::<Vec<_>>().connect(", ");
```

Comment:
Nit: please put the call to `.connect()` on the next line, lined up with the others in the chain